lib/puppet/settings.rb

```diff
@@ -1312,7 +1313,11 @@ def interpolate(name)
             raise InterpolationError, _("Error converting value for param '%{name}': %{detail}") % { name: name, detail: err }, err.backtrace
           end
 
-          setting.munge(val)
+          if to_string
+            interpolate_as_string(val, setting)
+          else
+            setting.munge(val)
+          end
```

Comment:
What about moving most of the `interpolate` logic to a private method (say `_interpolate` and then have two public methods `interpolate` and `print` (or `render` or ...?) that call the private method, eg

```ruby
def interpolate(name)
  setting, val = _interpolate(name)
  setting.munge(val)
end

def print(name)
  setting, val = _interpolate(name)
  setting.print(val)
end

def _interpolate(name)
  # does most of the work
  ...
  setting, val
end
```